libsolidity/codegen/CompilerContext.cpp

```diff
@@ -548,6 +548,7 @@ void CompilerContext::optimizeYul(yul::Object& _object, yul::EVMDialect const& _
 		_object,
 		_optimiserSettings.optimizeStackAllocation,
 		_optimiserSettings.yulOptimiserSteps,
+		isCreation? nullopt : make_optional(_optimiserSettings.expectedExecutionsPerDeployment),
```

Comment:
Why nullopt instead of 1?

Reply:
I think it's not enough to treat creation context as having `expectedExecutionsPerDeployment` as 1. This is because in creation context data gas is cheaper than at runtime context.